ToDoApp/ToDoApp/SignUp/SignUpViewModel.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  SignUpViewModel.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-20.
+//
+
+import Foundation
+import FirebaseAuth
+enum SignUpErrors: Identifiable, Hashable {
+    var id: Int {
+        hashValue
+    }
+    case invalidEmail
+    case invalidPassword
+    case invalidName
+    case authenticationError
+}
+@MainActor
+class SignUpViewModel: ObservableObject {
+    // MARK: Internal Properties
+    @Published var error: SignUpErrors?
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var name: String = ""
+    @Published var isCheckBoxTapped: Bool = false
+    private let generateUser: GenerateUser = GenerateUser()
+    private func trimTextField() {
+        self.email = email.trimmingCharacters(in: .whitespacesAndNewlines)
+        self.password = password.trimmingCharacters(in: .whitespacesAndNewlines)
+        self.name = name.trimmingCharacters(in: .whitespacesAndNewlines)
+    }
+    func signUp() {
```

Comment:
internal method names of view model which are used for the view should be non aware of the logic and represent events:
```suggestion
    func signUpButtonTapped() {
```
this helps easier navigate through the code